### PR TITLE
NGG coding rework: Small rework of exporting primitive

### PR DESCRIPTION
**Pull request by amdrexu**
Add a new function to export passthrough primitive. This is more clear. Also, rename the primitive export function. The original names are not straightforward. This change also considers the new layout of primitive connectivity data of future generations.

### CODE REVIEW

**Comment by amdvlk-admin** [comment]
### Test summary for commit 1a65ba8a8b8af6d0fefdeecd31bc12b20c037385
<details><summary>CTS tests (Failed: 3/281981)</summary><ul><li>Built with version 1.3.0.0</li>
<details open><summary>Ubuntu 22.04, Navi3x</summary><ul>

<li>Passed: 38862/70496 (55.1%)</li>
<li>Failed: 0/70496 (0.0%)</li>
<li>Not Supported: 31634/70496 (44.9%)</li><li>Warnings: 0/70496 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 22.04, Navi2x</summary><ul>

<li>Passed: 39109/70496 (55.5%)</li>
<li><details><summary>Failed: 3/70496 (0.0%)</summary><p>


Failures:
```
FAILURE: dEQP-VK.rasterization.conservative.overestimate.samples_4.triangles.degenerate.0_00
Stack trace: Fail

FAILURE: dEQP-VK.rasterization.conservative.overestimate.samples_4.triangles.degenerate.max
Stack trace: Fail

FAILURE: dEQP-VK.rasterization.conservative.overestimate.samples_4.triangles.degenerate.min
Stack trace: Fail
```
</p></details></li>

<li>Not Supported: 31384/70496 (44.5%)</li><li>Warnings: 0/70496 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 20.04, Gfx10</summary><ul>

<li>Passed: 35959/70498 (51.0%)</li>
<li>Failed: 0/70498 (0.0%)</li>
<li>Not Supported: 34539/70498 (49.0%)</li><li>Warnings: 0/70498 (0.0%)</li>
</ul></details>

<details open><summary>Rhel 9.0, Gfx9</summary><ul>

<li>Passed: 35861/70491 (50.9%)</li>
<li>Failed: 0/70491 (0.0%)</li>
<li>Not Supported: 34630/70491 (49.1%)</li><li>Warnings: 0/70491 (0.0%)</li>
</ul></details>


</ul></details>

File: lgc/patch/NggPrimShader.cpp [hunk 1]
```diff
@@ -2384,97 +2385,114 @@ void NggPrimShader::sendGsAllocReqMessage() {
...
+                                m_builder.getTrue(),  // done, must be set
+                                m_builder.getFalse(), // vm
+                            });
+}
 
-    if (m_compactVertex) {
-      auto compactVertIdBlock = createBlock(expPrimBlock->getParent(), ".compactVertId");
-      compactVertIdBlock->moveAfter(expPrimBlock);
+// =====================================================================================================================
+// Export primitive in culling mode with API GS.
```
Comment (nhaehnle) [inline lgc/patch/NggPrimShader.cpp:2406]: The comment is wrong?

**Review by nhaehnle — APPROVED** [review]
LGTM

File: lgc/patch/NggPrimShader.cpp (hunk 1 above)
Comment (amdrexu) [reply lgc/patch/NggPrimShader.cpp:2406]: Oh. My carelessness :)

**Comment by amdvlk-admin** [comment]
### Test summary for commit 43f8fa0870c284ad2c7a9ffad8f617c9e0d58459
<details><summary>CTS tests (Failed: 2/281981)</summary><ul><li>Built with version 1.3.0.0</li>
<details open><summary>Ubuntu 22.04, Navi3x</summary><ul>

<li>Passed: 38862/70496 (55.1%)</li>
<li>Failed: 0/70496 (0.0%)</li>
<li>Not Supported: 31634/70496 (44.9%)</li><li>Warnings: 0/70496 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 22.04, Navi2x</summary><ul>

<li>Passed: 39110/70496 (55.5%)</li>
<li><details><summary>Failed: 2/70496 (0.0%)</summary><p>


Failures:
```
FAILURE: dEQP-VK.rasterization.conservative.overestimate.samples_1.triangles.degenerate.max
Stack trace: Fail

FAILURE: dEQP-VK.rasterization.conservative.overestimate.samples_1.triangles.degenerate.min
Stack trace: Flake
```
</p></details></li>

<li>Not Supported: 31384/70496 (44.5%)</li><li>Warnings: 0/70496 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 20.04, Gfx10</summary><ul>

<li>Passed: 35959/70498 (51.0%)</li>
<li>Failed: 0/70498 (0.0%)</li>
<li>Not Supported: 34539/70498 (49.0%)</li><li>Warnings: 0/70498 (0.0%)</li>
</ul></details>

<details open><summary>Rhel 9.0, Gfx9</summary><ul>

<li>Passed: 35861/70491 (50.9%)</li>
<li>Failed: 0/70491 (0.0%)</li>
<li>Not Supported: 34630/70491 (49.1%)</li><li>Warnings: 0/70491 (0.0%)</li>
</ul></details>


</ul></details>

**Comment by amdvlk-admin** [comment]
### Test summary for commit 43f8fa0870c284ad2c7a9ffad8f617c9e0d58459
<details><summary>CTS tests (Failed: 0/281981)</summary><ul><li>Built with version 1.3.0.0</li>
<details open><summary>Ubuntu 22.04, Navi3x</summary><ul>

<li>Passed: 38862/70496 (55.1%)</li>
<li>Failed: 0/70496 (0.0%)</li>
<li>Not Supported: 31634/70496 (44.9%)</li><li>Warnings: 0/70496 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 22.04, Navi2x</summary><ul>

<li>Passed: 39112/70496 (55.5%)</li>
<li>Failed: 0/70496 (0.0%)</li>
<li>Not Supported: 31384/70496 (44.5%)</li><li>Warnings: 0/70496 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 20.04, Gfx10</summary><ul>

<li>Passed: 35959/70498 (51.0%)</li>
<li>Failed: 0/70498 (0.0%)</li>
<li>Not Supported: 34539/70498 (49.0%)</li><li>Warnings: 0/70498 (0.0%)</li>
</ul></details>

<details open><summary>Rhel 9.0, Gfx9</summary><ul>

<li>Passed: 35861/70491 (50.9%)</li>
<li>Failed: 0/70491 (0.0%)</li>
<li>Not Supported: 34630/70491 (49.1%)</li><li>Warnings: 0/70491 (0.0%)</li>
</ul></details>


</ul></details>

**Review by nhaehnle — APPROVED** [review]

**Comment by amdvlk-admin** [comment]
### Test summary for commit d7e05034198aba5492c2c13f3a0534f0e90de0e8
<details><summary>CTS tests (Failed: 0/282025)</summary><ul><li>Built with version 1.3.0.0</li>
<details open><summary>Ubuntu 22.04, Navi3x</summary><ul>

<li>Passed: 38862/70507 (55.1%)</li>
<li>Failed: 0/70507 (0.0%)</li>
<li>Not Supported: 31645/70507 (44.9%)</li><li>Warnings: 0/70507 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 22.04, Navi2x</summary><ul>

<li>Passed: 39112/70507 (55.5%)</li>
<li>Failed: 0/70507 (0.0%)</li>
<li>Not Supported: 31395/70507 (44.5%)</li><li>Warnings: 0/70507 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 20.04, Gfx10</summary><ul>

<li>Passed: 35959/70509 (51.0%)</li>
<li>Failed: 0/70509 (0.0%)</li>
<li>Not Supported: 34550/70509 (49.0%)</li><li>Warnings: 0/70509 (0.0%)</li>
</ul></details>

<details open><summary>Rhel 9.0, Gfx9</summary><ul>

<li>Passed: 35861/70502 (50.9%)</li>
<li>Failed: 0/70502 (0.0%)</li>
<li>Not Supported: 34641/70502 (49.1%)</li><li>Warnings: 0/70502 (0.0%)</li>
</ul></details>


</ul></details>

**Pull request merged**